src/securityPolicy.c

```diff
@@ -366,8 +386,9 @@ security_policy_t policyForSignTxInit(
 	WARN_UNLESS(isTxNetworkIdVerifiable(includeNetworkId, numOutputs, numWithdrawals, txSigningMode));
 	WARN_UNLESS(isNetworkUsual(networkId, protocolMagic));
 
-	WARN_IF(needsRunningScriptWarning(numCollaterals));
-	WARN_IF(needsMissingCollateralWarning(txSigningMode, numCollaterals));
+	WARN_IF(needsRunningScriptWarning(numCollateralInputs));
+	WARN_IF(needsMissingCollateralWarning(txSigningMode, numCollateralInputs));
```

Comment:
I'd rather not move them because the functions are called from UI machines in sign tx for all signing modes.

And yes, it is duplicated. There is a weakness in the design of the security policy return value mechanism. Originally, at most 1 warning was displayed, as the first step of the state machine. That is no longer true, but the return value does not convey the reason for warning, and the ui state machines have to be linear (because of how the macros are written), and now it is not possible to avoid displaying the warning just by skipping the first ui step because there might be multiple warnings to show. Not easy to improve, I'll create an issue.